chain/txValidator.go

```diff
@@ -157,6 +157,28 @@ func CheckTransactionPayload(txn *Transaction) error {
 		if subscriptionCount >= SubscriptionsLimit {
 			return errors.New(fmt.Sprintf("subscribtion count to %s can't be more than %d", pld.Topic, subscriptionCount))
 		}
+	case NanoPaymentChannelType:
+		pld := payload.(*NanoPaymentChannel)
+		if len(pld.ChannelId) > 32 {
+			return errors.New("length of channelId error")
+		}
+
+		if len(pld.Sender) != 20 && len(pld.Recipient) != 20 {
+			return errors.New("length of programhash error")
+		}
+
+		donationProgramhash, _ := ToScriptHash(config.DonationAddress)
```

Comment:
is the sender need to be donationaddress?

Reply:
it doesn't, so this check prevents it

Reply:
how to prevent multiple persons to use this channel id?

Reply:
why? it should be unique only per receiver

Reply:
What about make it unique per sender so that it can be easily extended to multiple receiver?

Reply:
resolving, since we moved to global state